### DISCUSSION TITLE
Distinguish between touching a chart and scrolling, highlighting a column

### DISCUSSION THREAD

**Discussion by andrey-ananiev** [patrykandpatrick/vico#465] [Ideas]
I want to touch the chart to get the index of the column, select it and save the selection even after touching. At the same time, so that scrolling does not highlight the columns. Unfortunately, I can't do this. It would be great if such an opportunity appeared.

**Comment by patrickmichalik** [comment]
Hello! Thanks for the suggestion. We’ll be looking into making this possible.

**Comment by JFBB** [comment]
Hello Patrick, have you managed to implement this yet?
I'm currently using 2.0.0-alpha20 with views

**Reply by Gowsky** [reply]
Hello, @JFBB. It's not been introduced yet. We're prioritizing the most-requested enhancements. Once this has been introduced, an update will be shared here.

Please note that version `2.0.0-alpha20` is quite outdated. It was released in May 2024.

**Comment by matejbart** [comment]
I'm voting for this feature as well.

In particular, it would be useful to have a touch listener for every column/component which also passes the coordinates of where exactly the column was touched.

For example, in a bar chart I would like to detect if the user touched the visible bar, and not just the entire reserved space for the column. As of v2.1.3, the only workaround is to use a marker listener, but this has the disadvantage that it relies on the internal visibility logic of the markers and you cannot determine the exact position as already noted here https://github.com/patrykandpatrick/vico/discussions/860.

**Comment by Gowsky** [comment]
[Vico v2.3.0-alpha.2](https://github.com/patrykandpatrick/vico/releases/tag/v2.3.0-alpha.2) introduces `CartesianMarkerController`.
